src/test/java/com/ncc/aif/TestUtils.java

```diff
@@ -0,0 +1,495 @@
+        Resource entity = makeEntity(model, getEntityUri(), system);
+        markJustification(addType(entity, type), getTypeAssertionJustification());
+        Resource entityCluster = makeClusterWithPrototype(model, getClusterUri(), entity, system);
+        return new Pair<>(entity, entityCluster);
+    }
+
+    /**
+     * Makes and returns a valid NIST-restricted event of the specified type and its cluster.
+     *
+     * @param type event type
+     * @return a key-value Pair of the event Resource (key) and its associated cluster Resource (value)
+     */
+    Pair<Resource, Resource> makeValidNistEvent(Resource type) {
+        Resource event = makeEvent(model, getEventUri(), system);
+        markJustification(addType(event, type), getTypeAssertionJustification());
+        Resource entityCluster = makeClusterWithPrototype(model, getClusterUri(), event, system);
+        return new Pair<>(event, entityCluster);
+    }
+
+    /**
+     * Makes and returns a valid NIST-restricted relation of the specified type and its cluster.
+     *
+     * @param type relation type
+     * @return a key-value Pair of the relation Resource (key) and its associated cluster Resource (value)
+     */
+    Pair<Resource, Resource> makeValidNistRelation(Resource type) {
+        Resource relation = makeRelation(model, getEventUri(), system);
+        markJustification(addType(relation, type), getTypeAssertionJustification());
+        Resource relationCluster = makeClusterWithPrototype(model, getClusterUri(), relation, system);
+        return new Pair<>(relation, relationCluster);
+    }
+
+}
+
+
+/**
+ * An extension of the NistTestUtils that supports testing TA3 restricted AIF.
+ * Usage and features are the same as NistTestUtils.
+ */
+class NistHypothesisTestUtils extends NistTestUtils {
+
+    /**
+     * Constructor for utilities for testing restricted AIF functionality.
+     *
+     * @param annotationNamespace namespace to use with URIs
+     * @param validator           an AIF validator instantiated based on the caller's ontology and desired NIST restrictions
+     * @param forceDump           whether or not to force dumping of models prior to validation
+     */
+    NistHypothesisTestUtils(String annotationNamespace, ValidateAIF validator, boolean forceDump) {
+        super(annotationNamespace, validator, forceDump);
+    }
+
+    /**
+     * Makes and returns a valid NIST-restricted entity of the specified type and its cluster with the specified
+     * cluster handle.
+     *
+     * @param type          entity type
+     * @param clusterHandle cluster handle for the entity cluster
+     * @return a key-value Pair of the entity Resource (key) and its associated cluster Resource (value)
+     */
+    Pair<Resource, Resource> makeValidNistEntity(Resource type, String clusterHandle) {
+        Pair<Resource, Resource> pair = makeValidNistEntity(type);
+        pair.getValue().addProperty(AidaAnnotationOntology.HANDLE, clusterHandle);
```

Comment:
Is there a reason that the handles and importances are explicit rather than some made up string/number? The main concern I have with this is that the normal makeValidNistEntity(Resource) won't create even a default handle/importance (and would thus not be valid). Let me know what you think

Reply:
Yeah, I'm glad you brought this up, because it's actually a symptom of a larger problem with the test utils.  You're right-- `NistHypothesisTestUtils` should overload `makeValidNistEntity(Resource)` to add an automatic/implicit handle and likewise `makeValidNistEvent` and `makeValidNistRelation` should overload and add an importance.

The "problem" is that it's really useful for the TA3 invalid tests to be able to create these things without the handles or importances.  See `NistHypothesisExamplesAndValidationTest::EntityClusterRequiredHandle.invalidNoHandle()` and `HypothesisEventRelationClusterImportanceValue` class.  They use the `NistTestUtils` version of makeValidEntity/Event/Relation to create these things without handles and importances, then add handles and importances as necessary to show the different invalid cases.

In general, we've created these TestUtiils so you can write the `valid()` methods very easily, but without a lot of help to the numerous `invalidXXX()` methods.  My "trick" in the above paragraph lets me have my cake and eat it too.

HOWEVER, this approach is used sporadically, and you are left with the unfortunate/bad situation where a user of the `NistHypothesisTestUtils` can create an *invalid* object using a `makeValidXXX()` method call, if they don't notice that the call comes from the parent `NistTestUtils` class (that is, if they don't add the extra `handle` or `importance` parameter).

At this stage in the game, I'm not sure what's best.  What you suggest is the pure approach, and I'd go back and update those invalid tests to create their objects mostly from scratch so that we can do everything validly except the handle/importance.  Another approach could be to rename the `NistHypoTestUtils` methods to `makeValidNistTA3XXX()`, so that you can call the TA3 version by default but still have access to the `makeValidNistXXX()` call that will be valid for Nist but not for Nist-TA3.  Ugh.

Reply:
Also note that `NistExamplesAndValidationTest.:JustifyTypeAssertions.invalid()` uses `TestUtil`'s `makeValidEntity/Event/Relation()` methods to create "valid" objects that don't have justified type assertions.

Reply:
This might warrant more discussion, but I was thinking we could use what we could of the super class's functionality to create the invalid objects. Let me know if you want to discuss this Monday or some other time

Reply:
Let's discuss Monday.

Reply:
After discussion, we decided that a sufficient change, for now at least, is to rename the methods so that it's more explicit what you're getting, without relying on overloading methods with additional parameters.  For example, `getValidAIFEntity()` returns a valid base AIF entity, `getValidNistEntity()` returns a valid NIST-restricted entity (it is clustered and the type assertion is justified), and `getValidTA3Entity()` returns a valid NIST-restricted entity for use in TA3 (the cluster is given an importance).

Class comments were updated to reflect this.